### PR TITLE
fortio 1.21.1

### PR DESCRIPTION
**Pull request by ldemailly**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
@ldemailly anything changed in the build, looks like the version test failed.

**Comment by chenrui333** [comment]
Actually it works fine in my local. Let's have a rerun (could be CI hiccup)

**Comment by ldemailly** [comment]
for 1.21.0 one of the test failed somehow showing 1.20.0 (previous version) this time it's dev (which is what happens when not building official-build) something is odd ...

**Comment by cho-m** [comment]
Probably a parallelization issue.

Failing CI runners:
```
cat: /tmp/fortio_build/link-flags.txt: No such file or directory
```

Passing CI runners:
```
mkdir -p /tmp/fortio_build
echo "2022-03-04 23:27 7965771ec8feec9de2e1c6a3752c594aee7350b0" > /tmp/fortio_build/build-info.txt
echo "-s -X main.defaultDataDir=. \
  -X \"fortio.org/fortio/version.buildInfo=2022-03-04 23:27 7965771ec8feec9de2e1c6a3752c594aee7350b0\" \
  -X fortio.org/fortio/version.version=1.21.1" | tee /tmp/fortio_build/link-flags.txt
-s -X main.defaultDataDir=.   -X "fortio.org/fortio/version.buildInfo=2022-03-04 23:27 7965771ec8feec9de2e1c6a3752c594aee7350b0"   -X fortio.org/fortio/version.version=1.21.1
```

**Comment by ldemailly** [comment]
Did you guys changed something somewhat recently because I haven't changed the build in years and it only started having issues in the last 2 releases (afaik). This being said if I need to change something to my Makefile to avoid these, happy to

**Comment by ldemailly** [comment]
btw the temp dir used is meant to be used inside docker environment (so in these case there isn't 2 build at the same time sharing a filesystem) but I made it relative to current dir which hopefully makes it unique (if that doesn't work we can use mktemp)

**Comment by cho-m** [comment]
That will reduce chance of issue and remove some problematic interleaving, though some race conditions still exists with Homebrew's default parallel `make -j<cpus>`.

First, `make` can simultaneously run `official-build-clean` and `official-build-version`. Also, the following in Makefile may be run in parallel:
```make
official-build: clean-link-flags official-build-internal
```

So, it is possible that:
1. `official-build-internal` starts, creates the `link-flags.txt` file
2. `clean-link-flags` starts and deletes the file
3. `official-build-internal` continues and does `cat link-flags.txt` on non-existent file

---

The rough workaround is to serialize the build (`ENV.deparallelize`), but this will disable parallelization everywhere and may have some build-time expense.

I think trying to do something like:
1. use your change to relative build directory 
2. have way to build without any `rm` commands (Homebrew can guarantee the build directory contains a fresh clone/checkout of code at beginning of `install` section).
3. run the `make` target that just does the installation step

**Comment by ldemailly** [comment]
we can call internal, the only reason for the clean is to avoid stale/leftover generated files in case of repeated builds but that doesn’t apply to brew (even I am surprised make -j would parallelize that) - I could also make the dependency harder as file not being found should just error.

random question: how do I format (go fmt equiv for ruby?) so the bot stops complaining about my change :)?

File: Formula/fortio.rb [hunk 1]
```diff
@@ -23,7 +23,8 @@ class Fortio < Formula
   depends_on "go" => :build
 
   def install
-    system "make", "official-build-clean", "official-build-version", "OFFICIAL_BIN=#{bin}/fortio"
+    system "make", "official-build-clean", "official-build-version", "OFFICIAL_BIN=#{bin}/fortio",
+	"BUILD_DIR=./tmp/fortio_build"
```
Comment (cho-m) [inline Formula/fortio.rb:27]: Maybe something like:

```suggestion
    system "make", "official-build-internal", "OFFICIAL_BIN=#{bin}/fortio", "BUILD_DIR=#{buildpath}/build"
```

Or some other directory name if `build` is already used. I didn't see anything that needed the extra nesting.

`#{buildpath}` should be absolute path to current PWD (top of cloned source tree).

Comment (ldemailly) [reply Formula/fortio.rb:27]: ./tmp works fine (and is in .gitignore) it's just that earlier the bot didn't like the line length and now doesn't like the tab

**Comment by ldemailly** [comment]
back to all green, thanks for the help @cho-m ! ptal

**Comment by ldemailly** [comment]
@chenrui333 all fixed up now, tia

**Review by SMillerDev — APPROVED** [review]
Thanks @ldemailly  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**